src/game/behaviors/fish.inc.c

```diff
@@ -1,163 +1,299 @@
+            schoolQuantity = 5;
+            minDistToMario = 1500.0f;
+            fishAnimation = cyan_fish_seg6_anims_0600E264;
             break;
     }
-    if (o->oDistanceToMario < sp30 || gCurrLevelNum == LEVEL_SA) {
-        for (i = 0; i < sp38; i++) {
-            sp28 = spawn_object(o, model, bhvFishGroup2);
-            sp28->oBehParams2ndByte = o->oBehParams2ndByte;
-            func_8029EE20(sp28, sp2C, 0);
-            translate_object_xyz_random(sp28, 700.0f);
+    /**
+     * If Distance to Mario is lower than 1500 or the current level is Secret Aquarium,
+     * then spawn and animate the schoolQuantity of fish.
+     * Fish moves at random with a max-range of 700.0f.
+     */
+    if (o->oDistanceToMario < minDistToMario || gCurrLevelNum == 20) {
+        for (i = 0; i < schoolQuantity; i++) {
+            fishObject = spawn_object(o, model, bhvFishGroup2);
+            fishObject->oBehParams2ndByte = o->oBehParams2ndByte;
+            func_8029EE20(fishObject, fishAnimation, 0);
+            translate_object_xyz_random(fishObject, 700.0f);
         }
-        o->oAction = 1;
+        o->oAction = FISH_ACTIVE;
     }
 }
 
-void ActionFish1(void) {
-    if (gCurrLevelNum != LEVEL_SA)
-        if (gMarioObject->oPosY - o->oPosY > 2000.0f)
-            o->oAction = 2;
+/**
+ * If the current level is not Secret Aquarium and the distance from
+ * mario->PosY minus o->oPosY is greater than 2000.0f then set oAction to FISH_RESPAWN.
+ */
+void fish_act_check_distance(void) {
+    if (gCurrLevelNum != 20) {
+        if (gMarioObject->oPosY - o->oPosY > 2000.0f) {
+            o->oAction = FISH_RESPAWN;
+        }
+    }
 }
 
-void ActionFish2(void) {
-    o->oAction = 0;
+/**
+ * set oAction flag to FISH_INIT.
+ */
+void fish_act_set(void) {
+    o->oAction = FISH_INIT;
 }
 
-void (*sFishActions[])(void) = { ActionFish0, ActionFish1, ActionFish2 };
+/**
+ * An array of action methods
+ */
+void (*sFishActions[])(void) = { fish_act_spawn, fish_act_check_distance, fish_act_set };
 
+/**
+ * Calls an array of actions based on the value of o->oAction
+ */
 void bhv_fish_loop(void) {
     obj_call_action_function(sFishActions);
 }
 
-void func_802BF59C(s32 a0) {
-    f32 sp1C = o->parentObj->oPosY;
-    if (gCurrLevelNum == LEVEL_SA) {
-        if (500.0f < absf(o->oPosY - o->oFishGroupUnkF8))
-            a0 = 10;
-        o->oPosY = approach_f32_symmetric(o->oPosY, o->oFishGroupUnkF8, a0);
-    } else if (sp1C - 100.0f - o->oFishGroupUnk10C < o->oPosY
-               && o->oPosY < sp1C + 1000.0f + o->oFishGroupUnk10C)
-        o->oPosY = approach_f32_symmetric(o->oPosY, o->oFishGroupUnkF8, a0);
+/**
+ * Move object up or down towards oFishPosY by the speed variable
+ * In Secret Aquarium, the speed variable is always ten.
+ */
+void fish_regroup(s32 speed) {
+    // Set pPosY to the parentY.
+    f32 pPosY = o->parentObj->oPosY;
+    
+    // If level is SA and fish height minus oPosY is higher than 500 then speed = 10. 
```

Comment:
You're describing what the code does, but not *why* it does it. This is also an issue in lots of other places in the files, but it's more of an abstract issue so it's hard to show an outright example.